applications/plugins/SofaPython/Binding_GUIManager.cpp

```diff
@@ -0,0 +1,249 @@
+#include "Binding_GUIManager.h"
+#include "PythonMacros.h"
+#include "PythonToSofa.inl"
+#include <sofa/gui/GUIManager.h>
+#include <sofa/gui/BaseGUI.h>
+#include <sofa/simulation/Node.h>
+
+static constexpr const char* listSupportedGUI_DOC =
+R"DOC(
+Lists the supported GUI types.
+
+:return: A list of strings populated with the supported GUI types
+:rtype: list
+)DOC";
+static PyObject * GUIManager_listSupportedGUI(PyObject * /*self*/, PyObject * /*args*/) {
+    const std::vector<std::string> guiList = sofa::gui::GUIManager::ListSupportedGUI();
+    PyObject * list = PyList_New(guiList.size());
+    for (unsigned int i = 0; i < guiList.size(); ++i) {
+        PyObject *gui = PyString_FromString(guiList[i].c_str());
+        PyList_SetItem(list, i, gui);
+    }
+    return list;
+}
+
+
+static constexpr const char* setConfigDirectoryPath_DOC =
+R"DOC(
+Set the Sofa's configuration directory path.
+
+:param path: The path to the Sofa's configuration directory
+:type path: str
+)DOC";
+static PyObject * GUIManager_setConfigDirectoryPath(PyObject * /*self*/, PyObject * args) {
+    const char * path_ptr;
+    PyObject * createIfNecessaryObj = nullptr;
+
+    if (!PyArg_ParseTuple(args, "s|O", &path_ptr, &createIfNecessaryObj)) {
+        SP_MESSAGE_ERROR("GUIManager.setConfigDirectoryPath requires a path as first argument.");
```

Comment:
If I understand, the function is throwing an exception. The exception's message is set in the PyArg_ParseTuple and 
and additional message is printed with SP_MESSAGE_ERROR (so going through the the message pipeline). 
Wouldn't it be better to only have one message, the one associated with the python exception and set using the python API ?